lib/helpers/resolve-files.js

```diff
@@ -40,13 +42,18 @@ const pathUntilGlob = (path) => {
 const resolveFiles = async (files, cwd) =>
     Promise.all(
         Object.entries(files).map(async (definition) => {
-            const [, source] = definition;
+            let [, source] = definition;
+
+            // The config may not always match the OS's separator.
+            // Convert the input to OS-specific if necessary.
+            source = ensureOsSep(source);
```

Comment:
Might want to explain in the comment why we bother doing this when we convert back later (it's so the Node path functions don't get confused when we're actually building an OS file path)